pkg/util/toggle.go

```diff
@@ -0,0 +1,24 @@
+package util
+
+import "sync"
+
+type Toggle struct {
+	enabled bool
+	mux     sync.RWMutex
+}
+
+func NewToggle() *Toggle {
```

Comment:
Can you add some comments around the functionality this serves

Reply:
done